Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -407,12 +428,16 @@ void BlockFormattingContext::layout_block_level_children(BlockContainer const& b
     VERIFY(!block_container.children_are_inline());
 
     float bottom_of_lowest_margin_box = 0;
+    float current_y = 0;
 
     block_container.for_each_child_of_type<Box>([&](Box& box) {
-        layout_block_level_box(box, block_container, layout_mode, bottom_of_lowest_margin_box, available_space);
+        layout_block_level_box(box, block_container, layout_mode, bottom_of_lowest_margin_box, available_space, current_y);
         return IterationDecision::Continue;
     });
 
+    // FIXME: margin-bottom of last in-flow child can be collapsed with margin-bottom of parent
```

Comment:
Would be great to implement this as well in the same PR, so we see that the `BlockMarginState` concept works for all situations.